packages/sitecore-jss-react/src/enhancers/withEmptyValueEditingPlaceholder.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { ComponentType, forwardRef } from 'react';
+import { LinkFieldValue } from '../components/Link';
+import { ImageFieldValue } from '../components/Image';
+
+interface GeneralField {
+  metadata?: { [key: string]: unknown };
+  value?: unknown;
+}
+
+interface WithEmptyValueEditingPlaceholderProps {
+  field?: GeneralField;
+  editable?: boolean;
+  emptyValueEditingPlaceholder?: React.ComponentClass | React.FC;
+}
+
+/**
+ * Returns the passed field component or placeholder content in case field value is empty and edit mode is 'metadata'
+ * @param {ComponentType<FieldComponentProps>} FieldComponent the field component
+ * @param {React.FC} defaultEmptyFieldEditingComponent the default empty field placeholder component
+ * @param {boolean} isForwardRef set to 'true' if forward reference is needed
+ */
+export function withEmptyValueEditingPlaceholder<
+  FieldComponentProps extends WithEmptyValueEditingPlaceholderProps,
+  RefElementType = HTMLElement
+>(
+  FieldComponent: ComponentType<FieldComponentProps>,
+  defaultEmptyFieldEditingComponent: React.FC,
```

Comment:
if empty component is being passed into HOC anyway, would it be better to combine this and the emptyValueEditingPlaceholder props into one? Just have the field components decide which implementation to use and pass into HOC?

Reply:
I agree. I think it's cleaner to have HOC take a single required prop (remove any knowledge of default) and the consumers should handle sending override or default (e.g. `emptyValueEditingPlaceholder || DefaultEmptyFieldEditingComponentText`).

Reply:
that makes sense. But seems we can't pass the default value via props - see what we need to do for editable here: https://github.com/Sitecore/jss/blob/4a047aa201182c8a0b678f949f8fb9f72c6a7783/packages/sitecore-jss-react/src/enhancers/withFieldMetadata.tsx#L40